src/Lib/Buddy.php

```diff
@@ -0,0 +1,245 @@
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License version 2 or any later
+  version. You should have received a copy of the GPL license along with this
+  program; if you did not, you can find it at http://www.gnu.org/
+*/
+
+namespace Manticoresearch\Buddy\Lib;
+
+use Manticoresearch\Buddy\Enum\ACTION;
+use Manticoresearch\Buddy\Interface\CustomErrorInterface;
+use Manticoresearch\Buddy\Interface\QueryParserLocatorInterface;
+use Manticoresearch\Buddy\Interface\StatementBuilderInterface;
+use Manticoresearch\Buddy\Trait\CheckClientStateTrait;
+use Manticoresearch\Buddy\Trait\CustomErrorTrait;
+use Throwable;
+
+class Buddy {
+
+	use CustomErrorTrait;
+	use CheckClientStateTrait;
+
+	const ACTION_MAP = [
+		'NO_INDEX' => [
+			'INSERT_QUERY' => [ACTION::CREATE_INDEX, ACTION::INSERT],
+		],
+		'UNKNOWN_COMMAND' => [
+			'SHOW_QUERIES_QUERY' => [ACTION::SELECT_SYSTEM_SESSIONS],
+		],
+	];
+
+	/**
+	 * @var string $query
+	 */
+	protected string $query;
+	/**
+	 * @var string $msg
+	 */
+	protected string $msg;
+	/**
+	 * @var string $origMsg
+	 */
+	protected string $origMsg;
+	/**
+	 * @var string $errorMsg
+	 */
+	protected string $errorMsg;
+	/**
+	 * @var string $reqFormat
+	 */
+	protected string $reqFormat;
+
+	/**
+	 * @param string $clAddrPort
+	 * @param QueryParserLocatorInterface|null $queryParserLocator
+	 * @param StatementBuilderInterface|null $stmtBuilder
+	 * @param CustomErrorInterface|null $exceptionHandler,
+	 * @return void
+	 */
+	public function __construct(
+		protected string $clAddrPort,
+		protected QueryParserLocatorInterface $queryParserLocator = null,
+		protected StatementBuilderInterface $stmtBuilder = null,
+		protected CustomErrorInterface $exceptionHandler = null,
+	) {
+		$this->msg = $this->errorMsg = $this->origMsg = '';
+	}
+
+	/**
+	 * @param string $origMsg
+	 * @param string $query
+	 * @param string $reqFormat
+	 * @return array{type:string, message:string, error:string}
+	 */
+	public function getResponse(string $origMsg, string $query, string $reqFormat): array {
+		$this->origMsg = $origMsg;
+		$this->query = $query;
+		$this->reqFormat = $reqFormat;
+		$this->msg = $this->errorMsg = '';
+		$handleActions = $this->getHandleActions();
+		if (!empty($handleActions)) {
+			try {
+				// sending the chain of needed requests to Manticore
+				foreach ($handleActions as $action) {
+					$mntRequest = $this->buildMntRequest($action);
+					$this->msg = $this->getMntResponse($mntRequest);
+					// checking if any of requests fails
+					if ($this->isMntResponseOk() === false) {
+						$this->errorMsg = $this->msg;
+						break;
+					}
+				}
+			} catch (Throwable $e) {
+				$this->errorMsg = $e->getMessage();
+			}
+		}
+
+		return $this->buildResponse();
+	}
+
+	/**
+	 * @param string $message
+	 * @param string $error
+	 * @return array{type:string, message:string, error:string}
```

Comment:
This is a topic to discuss, I guess,

I'm using VSCode and syntax with space not highlighted, so I made declarations like

`@return array{type:string,message:string,error:string}`

I'm not sure what I can do with it in VSCode, but it has no highlight at all when it has space.

This is the topic to discuss